include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -524,7 +528,11 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
                                           std::vector<NodeID> &packed_path) const
     {
         NodeID current_node_id = middle_node_id;
-        while (current_node_id != search_heap.GetData(current_node_id).parent)
+        // all initial nodes will have itself as parent
+        // all incomplete paths (e.g. paths ending at the core boundary) will have SPECIAL_NODEID
+        // as parent
+        while (current_node_id != search_heap.GetData(current_node_id).parent &&
+               search_heap.WasInserted(search_heap.GetData(current_node_id).parent))
```

Comment:
I don't get why we need this two cases. Was there anything breaking when adding the core nodes with themselves as a parent? 

From looking at [this](https://github.com/Project-OSRM/osrm-backend/blob/cb57ccacefa59321b9a921a760fa70b08d49d30d/include/engine/routing_algorithms/routing_base.hpp#L712) we where using the node without a dedicated parent. So while unpacking the path, we should stop at each core node [here](https://github.com/Project-OSRM/osrm-backend/blob/cb57ccacefa59321b9a921a760fa70b08d49d30d/include/engine/routing_algorithms/routing_base.hpp#L527)

I fail to see how this modification actually changes something, other than adding to the `if` condition here. The results should just be the same?